docs/user/debugging.md

```diff
@@ -71,28 +83,143 @@ More information can be found in the [View Logs with Logcat][] help article.
 
 [pidcat]: https://github.com/JakeWharton/pidcat
+
+## Debugging the Glean SDK in iOS
+
+For debugging and validation purposes on iOS, Glean makes use of a custom URL scheme which is implemented _within the application_ that is consuming Glean.  Glean provides some convenience functions to facilitate this, but it's up to the consuming application to enable this functionality.  Applications that enable this Glean SDK feature will be able to launch the application from a URL with the Glean debug commands embedded in the URL itself.
+
+### Available commands and query format
+
+There are 3 available commands that you can use with the Glean SDK debug tools
+
+- `logPings`: This is either true or false and will cause pings that are submitted to also be echoed to the device's log
+- `tagPings`: This command expects a string with upper and lower case letters, numbers and dashes, with a max length of 20 characters that will be used to tag the ping and cause it to appear in the Glean Debug View.
+- `sendPing`: This command expects a string name of a ping to force immediate collection and submission of.
+
+The structure of the custom URL uses the following format:
+
+`<protocol>://glean?<command 1>=<paramter 1>&<command 2>=<parameter 2> ...`
+
+Where:
+
+- `<protocol>` is the "Url Scheme" that has been added for your app (see Instrumenting the application below), such as `glean-sample-app`.
+- This is followed by `://` and then `glean` which is required for the Glean SDK to recognize the command is meant for it to process.
+- Following standard URL query format, the next character after `glean` is the `?` indicating the beginning of the query.
+- This is followed by one or more queries in the form of `<command>=<parameter>`, where the command is one of the commands listed above, followed by an `=` and then the value or parameter to be used with the command.
+
+There are a few things to consider when creating the custom URL:  
+
+- Invalid commands will trigger an error and be ignored.
+- Not all commands are requred to be encoded in the URL, you can mix and match the commands that you need.  
+- Special characters should be properly URL encoded and escaped since this needs to represent a valid URL.
+
+### Instrumenting the application for Glean debug functionality
+
+In order to enable the debugging features in a Glean SDK consuming iOS application, it is necessary to add some information to the application's `Info.plist`, and add a line and possibly an override for a function in the `AppDelegate.swift`.
+
+#### Register custom URL scheme in `Info.plist`
+
+If your application already has a custom URL scheme implemented, there is no need to implement a second scheme, you can simply use that and skip to the next section about adding the convenience method.  If the app doesn't have a custom URL scheme implemented, then you will need to perform the following instructions to register your app to recieve custom URLs.
```

Comment:
```suggestion
> **Note:** If your application already has a custom URL scheme implemented, there is no need to implement a second scheme, you can simply use that and skip to the next section about adding the convenience method.  If the app doesn't have a custom URL scheme implemented, then you will need to perform the following instructions to register your app to recieve custom URLs.
```

Reply:
What do you think of the above? It will render as an "important" text box, to highlight it.